master/internal/resourcemanagers/priority_test.go

```diff
@@ -0,0 +1,240 @@
+package resourcemanagers
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+)
+
+func TestGetAllPendingZeroSlotTasks(t *testing.T) {
+	agents := make([]*mockAgent, 0)
+	groups := []*mockGroup{
+		{id: "group1", maxSlots: newMaxSlot(1), weight: 1},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0]},
+		{id: "task2", slotsNeeded: 1, group: groups[0]},
+		{id: "task3", slotsNeeded: 0, group: groups[0]},
+		{id: "task4", slotsNeeded: 0, group: groups[0]},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, _, _ := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	pendingZeroSlotTasks := getAllPendingZeroSlotTasks(taskList, "")
+	expectedZeroSlotTasksIds := []*mockTask{tasks[2], tasks[3]}
+	assertEqualToAllocate(t, pendingZeroSlotTasks, expectedZeroSlotTasksIds)
+
+	setTaskAllocations(t, taskList, "task4", 1)
+
+	pendingZeroSlotTasks = getAllPendingZeroSlotTasks(taskList, "")
+	expectedZeroSlotTasksIds = []*mockTask{tasks[2]}
+	assertEqualToAllocate(t, pendingZeroSlotTasks, expectedZeroSlotTasksIds)
+}
+
+func TestSortTasksByPriorityAndTimestamps(t *testing.T) {
+	lowerPriority := 50
+	higherPriority := 40
+
+	agents := make([]*mockAgent, 0)
+	groups := []*mockGroup{
+		{id: "group1", priority: &lowerPriority},
+		{id: "group2", priority: &higherPriority},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0]},
+		{id: "task2", slotsNeeded: 1, group: groups[0]},
+		{id: "task3", slotsNeeded: 0, group: groups[1]},
+		{id: "task4", slotsNeeded: 0, group: groups[1]},
+		{id: "task5", slotsNeeded: 4, group: groups[1]},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, mockGroups, _ := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	pendingTasksByPriority, _ := sortTasksByPriorityAndTimestamp(taskList, mockGroups, "")
+	for priority, tasksInPriority := range pendingTasksByPriority {
+		switch priority {
+		case lowerPriority:
+			expectedTasks := []*mockTask{tasks[0], tasks[1]}
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		case higherPriority:
+			expectedTasks := []*mockTask{tasks[4]}
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		default:
+			panic("unexpected priority")
+		}
+	}
+
+	setTaskAllocations(t, taskList, "task5", 1)
+
+	_, scheduledTasksByPriority := sortTasksByPriorityAndTimestamp(taskList, mockGroups, "")
+	for priority, tasksInPriority := range scheduledTasksByPriority {
+		switch priority {
+		case lowerPriority:
+			expectedTasks := make([]*mockTask, 0)
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		case higherPriority:
+			expectedTasks := []*mockTask{tasks[4]}
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		default:
+			panic("unexpected priority")
+		}
+	}
+}
+
+func TestPrioritySchedulingNoPreemption(t *testing.T) {
```

Comment:
q: by "no preemption", do you mean the scheduler is set not to preempt or the expected behavior is no preemption to happen?

Reply:
It looks the answer is the former. sorry I totally misunderstood the tests. However, we should change the name to be more detailed.

`TestPrioritySchedulingBasic` should be just fine.

Reply:
Updated this name lmk if you don't think it's clear.